openedx/core/djangoapps/user_api/profiles/views.py

```diff
@@ -0,0 +1,113 @@
+"""
+NOTE: this API is WIP and has not yet been approved. Do not use this API without talking to Christina or Andy.
+
+For more information, see:
+https://openedx.atlassian.net/wiki/display/TNL/User+API
+"""
+from django.conf import settings
+from django.contrib.auth.models import User
+
+from rest_framework import status
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework.authentication import OAuth2Authentication, SessionAuthentication
+from rest_framework import permissions
+
+from ..accounts.views import AccountView
+from ..api.account import AccountUserNotFound
+from ..models import UserPreference
+
+from . import PROFILE_VISIBILITY_PREF_KEY, ALL_USERS_VISIBILITY
+
+
+class ProfileView(APIView):
+    """
+        **Use Cases**
+
+            Get the user's public profile information.
+
+        **Example Requests**:
+
+            GET /api/user/v0/profiles/{username}/[?include_all={true | false}]
+
+        **Response Values for GET**
+
+            Returns the same responses as for the AccountView API, but
+            filtered based upon the user's specified privacy permissions.
+            If the parameter include_all is passed as 'true' then a user
+            can receive all fields for their own account, ignoring
+            the field visibility preference.
```

Comment:
I'd suggest adding that the include_all URL parameter is ignored if the profile requested is not for the user making the request.